graph/tree/tree.go

```diff
@@ -0,0 +1,38 @@
+// tree represents a parent-child tree and is used to represent a service
+// hierarchy. Root nodes have parent == nil.  Metadata is a general purpose map
+// for holding any desired service information. Each Service node should have
+// a unique name+version
+package tree
+
+import (
+	"fmt"
+)
+
+const (
+	UnknownName    = "internet"
+	UnknownVersion = "unknown"
+)
+
+type ServiceNode struct {
+	ID       string
```

Comment:
What's the difference between ID and Name ?
What are the sources of the mapping of these two fields ?

Reply:
Yes,  service name can have many nodes, one for each version.  ID is unique tuple.  I added some additional documentation and also removed the custom ID from the "constructor", it was left over from some earlier work.